hangman.py

```diff
@@ -1,213 +1,323 @@
+    """
+    if is_word_guessed(secret_word, letters_guessed):
+        victory(guesses, secret_word)
+        return True
+    elif guesses <= 0:
+        print(TextStyle.change_background_font(f'You lost. Guesses exceeded. The secret word was - {secret_word}'))
+        return True
+    return False
+
+
+def welcome_art(secret_word: str):
+    """
+    secret_word: the word the user is guessing
+    """
+    print('''
+     _   _                                           __   _____ 
+    | | | |                                         /  | |  _  |
+    | |_| | __ _ _ __   __ _ _ __ ___   __ _ _ __   `| | | |/' |
+    |  _  |/ _` | '_ \ / _` | '_ ` _ \ / _` | '_ \   | | |  /| |
+    | | | | (_| | | | | (_| | | | | | | (_| | | | | _| |_\ |_/ /
+    \_| |_/\__,_|_| |_|\__, |_| |_| |_|\__,_|_| |_| \___(_)___/ 
+                        __/ |                                   
+                       |___/                                    
+    ''')
+    print(f'I am thinking of a word that is {len(secret_word)} letters long.')
+
+
+def hangman(secret_word, hints=False):
+    '''
+    secret_word: string, the secret word to guess.
+    Starts up an interactive game of Hangman.
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+    * The user should start with 6 guesses
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    welcome_art(secret_word)
+
+    # start variables
+    guesses = GUESSES
+    letters_guessed = []
+    warnings = WARNINGS
+
+    # main game cycle
+    while 1:
+        result = game_result(guesses, letters_guessed)
+        if result:
+            break
+        else:
+            # let the user know his resources
+            print(f'You have {guesses} guesses left.')
+            available = get_available_letters(letters_guessed)
+            print(f'Available letters: {available}')
+
+            # validate letter
+            while 1:
+                letter_entered = input('Enter your guess letter: ')
+                letter, guesses, warnings = letter_check(letter_entered, guesses, warnings)
+                if letter != 'Invalid value':
+                    break
+            letters_guessed.append(letter.lower())
+
+            # check if user want hint
+            if hints:
+                if letter == STAR:
+                    show_possible_matches(get_guessed_word(secret_word, letters_guessed))

```

Comment:
I'd check for hints before letter validation and called `continue`, so that there will be no need to check in all validations whether user asked for hint, which should be a valid input

Reply:
Ok